flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableEnvironment.java

```diff
@@ -103,6 +106,23 @@ static TableEnvironment create(EnvironmentSettings settings) {
 	 */
 	Optional<Catalog> getCatalog(String catalogName);
 
+	/**
+	 * Loads a {@link Module} under a unique name. Modules will be kept in the loaded order.
+	 *
+	 * @param moduleName name of the {@link Module}
+	 * @param module the module instance
+	 * @throws ModuleAlreadyExistException thrown when there is already a module with the same name
+	 */
+	void loadModule(String moduleName, Module module) throws ModuleAlreadyExistException;
```

Comment:
Sorry, that's a point I missed considering in the FLIP but realized when implementing it.

After comparing with how other APIs, like catalog in TableEnv, handle exceptions, I agree we should remove the checked exception. Similar to `CatalogException`, how about we create a `ModuleException extends RuntimeException` and it will be used very similar to `CatalogException`?

Reply:
Do we really need a specific exception for it? Isn't `ValidationException` enough? What benefit do we get with a `ModuleException`? The `ValidationException`/`TableException` clearly separates user-side errors from system-side errors. If you really want to introduce additional exception let's try to maintain this distinction.

BTW, regarding the exceptions in Catalog API, I had a few discussions with @twalthr before and we both agree that at some point we should decrease the number of exceptions being thrown from the Catalog API. As of now it is not a very user friendly API in that regard. Therefore I would not take it as an example.